test/images/vulnerability-scan/scan.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+
+images=(
+ "access-setup"
+ "argocd-registrar"
+ "cluster-setup"
+ "gateway-deployment"
+ "kcp-registrar"
+ "shellcheck"
+)
+
```

Comment:
This means that we fetch the scan of every images when a PR modifies any of them.
Technically an image scan may return new vulnerabilities when the image itself has not been modified (due to new CVEs being added to Clair's vulnerability database) but I feel that it is not helpful in case of a pull request.
I would propose the following:
- to have a GitHub action that retrieves image scans only for images that have been modified by the PR, so that PR authors don't get asked to fix a vulnerability in an image they have not touched
- to have a periodic GitHub action that once a day retrieves the scan results for all the images.

Reply:
I have developed two GitHub actions now.
periodic-scanner-quay.yaml - Runs everyday at 00.00.00
individual-image-scanner-quay.yaml - Runs with every PR. But only scans for those images that are modified in the PR.